budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -168,3 +174,141 @@ def test_other_parties(self):
         self.assertEqual(
             other_parties, ["Service Provider: DOPW\nService Provider: AAAA"]
         )
+
+
+class ProvInfraProjectAPITestCase(APITestCase):
+    def setUp(self):
+        """Create 30 Provincial Infrastructure Projects"""
+
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.url = reverse("search-provincial-infrastructure-projects")
+        self.province = u"Eastern Cape"
+        self.status = u"Construction"
+        self.source = u"Community Library Service Grant"
+        self.department = u"Department 1"
+        self.name = u"Project 1"
+        self.municipality = u"Local 2"
+        self.contractor = u"Contractor 3"
+
+        self.provinces = [
+            "Eastern Cape",
+            "Free State",
```

Comment:
you don't need 9 provinces for tests - just use the minimal number needed to verify the functionality you're trying to verfy

Reply:
it decreased to 2.

https://github.com/vulekamali/datamanager/pull/190/commits/ee17bc29b03185e791fa6fa2001619b1076a4683